lib/api/core/models/security/role.js

```diff
@@ -352,3 +345,193 @@ Role.prototype.validateDefinition = function (context) {
 };
 
 module.exports = Role;
+
+
+/**
+ * @param {Kuzzle} kuzzle
+ * @param {Array} path
+ * @param {Object} actionRights
+ * @param {RequestObject} requestObject
+ * @param {Object} context
+ * @returns {Promise}
+ */
+function executeClosure (kuzzle, path, actionRights, requestObject, context) {
+  var
+    sandboxContext,
+    sandboxScript,
+    error,
+    argsDefinitions = {},
+    message;
+
+  if (this.closures[path] === undefined) {
+    this.closures[path] = {};
+  }
+
+  if (actionRights.args && !_.isEmpty(actionRights.args) && !this.closures[path].getArgsDefinitions) {
+    try {
+      /* jshint evil: true */
+      this.closures[path].getArgsDefinitions = global.eval('(function (requestObject) {return ' +
+        JSON
+          .stringify(actionRights.args)
+          .replace(/"\$(requestObject\.data\.[a-zA-Z0-9_\-.]*[a-zA-Z0-9])"/g, '$1')
+          .replace('"$currentId"', 'requestObject.data._id') +
+        ';})');
+      /* jshint evil: false */
+    }
+    catch (err) {
+      error = new InternalError('Error parsing rights for role ' + this._id + ' (' + path.join('/') + ') :' + actionRights);
```

Comment:
If I understand correctly, a parse error occurs if an administrator has configured an invalid closure.
If so, then instead of an `InternalError`, I would use either a new `ParseError` or a new `InvalidRightError` (or `InvalidClosureError`) error.
To me, `InternalError` should be reserved to unhandled Kuzzle errors